packages/util/src/menu-keydown-listener.js

```diff
@@ -0,0 +1,107 @@
+import { useCallback, createRef } from 'react'
+import { useCombinedRefs } from './use-combined-refs.js'
+
+const searchListItem = (el, down = true) => {
+  return el.tagName === 'LI'
+    ? el
+    : el.tagName !== 'LI'
+    ? searchListItem(
+        down ? el.nextElementSibling : el.previousElementSibling,
+        down
+      )
+    : false
+}
+
+const rightArrow = e => {
+  const subMenu = e.target.querySelector(':scope > ul')
+  if (subMenu.tagName !== 'UL') return
+  const firstChild = searchListItem(subMenu.firstElementChild)
+  firstChild && firstChild.focus()
+}
+
+const leftArrow = e => {
+  const rootMenu = e.target.parentNode === e.currentTarget
+  if (rootMenu) return
+  e.target.blur()
+  e.target.parentNode.parentNode.focus()
+}
+
+const downArrow = e => {
+  const nextSibling = searchListItem(e.target.nextElementSibling)
+  const firstChild = searchListItem(e.target.parentNode.firstElementChild)
+  nextSibling ? nextSibling.focus() : firstChild.focus()
+}
+
+const upArrow = e => {
+  const prevSibling = searchListItem(e.target.previousElementSibling, false)
+  const lastChild = searchListItem(e.target.parentNode.lastElementChild, false)
+  prevSibling ? prevSibling.focus() : lastChild.focus()
+}
+
+const home = e => {
+  const firstChild = searchListItem(e.target.parentNode.firstElementChild)
+  firstChild.focus()
+}
+
+const end = e => {
+  const lastChild = searchListItem(e.target.parentNode.lastElementChild, false)
+  lastChild.focus()
+}
+
+const startsWith = (el, key) => {
+  return (
+    el.innerText.startsWith(key) || el.innerText.startsWith(key.toUpperCase())
+  )
+}
+
+const char = e => {
```

Comment:
I really must not understand this function.  My current best guess: If the text of the sibling starts with the letter of the key pressed, then focus it.

?? Why, I'm not sure.

Reply:
*Moves focus to the next item having a name that starts with the typed character* 

It's not perfect in fact I think I messed up a bit on this w3 is not clear if it should search submenus so I set it to not search submenus as every other interaction happened on the same tree level.

Reply:
Wow, so I was reading it right.  Thanks for your other w3 link.  I didn't know this existed as a suggestino.

Yeah, makes sense to me to search on just the current level.